codec-http2/src/test/java/io/netty/handler/codec/http2/internal/hpack/DecoderTest.java

```diff
@@ -74,14 +81,63 @@ public void setUp() {
     }
 
     @Test
+    public void testLiteralHuffmanEncodedWithEmptyNameAndValue() throws IOException {
+        byte[] input = {0, (byte) 0x80, 0};
+        ByteArrayInputStream in = new ByteArrayInputStream(input);
+        try {
+            decoder.decode(in, mockListener);
+            verify(mockListener, times(1)).addHeader(EMPTY_BYTES, EMPTY_BYTES, false);
+        } finally {
+            in.close();
+        }
+    }
+
+    @Test(expected = IOException.class)
+    public void testLiteralHuffmanEncodedWithPaddingGreaterThan7Throws() throws IOException {
+        byte[] input = {0, (byte) 0x81, -1};
+        ByteArrayInputStream in = new ByteArrayInputStream(input);
```

Comment:
Could you separate out this into a separate utility method to clean up the test methods? It would be easier to read if we had something like:

```java
byte[] input = ...
decode(input);
verify(mockListener, times(1))...
```